### ISSUE TITLE
Playlist randomizer is really bad

### ISSUE THREAD

**Issue by emoon** [emoon/HippoPlayer#89]
* Can replay the same song several times
* Go back doesn't actually track the previous song
* etc

**Comment by sagamusix** [comment]
Food for thought, may or may not be applicable to Hippo:

XMPlay has separate concepts of library and playlist. Until recently the randomizer only took the playlist into account - songs were marked as played and then couldn't be chosen by the randomizer again until all songs were played. So far, so obvious.

However, if you reload the playlist, the played status gets reset for all files obviously. I sometimes need to reload my playlist, and I noticed that as a consequence, I had songs that haven't been played *in years*. So I suggested that XMPlay should have an option to favor songs that haven't been played in a long time (according to the "last played" field in the library, which does not get reset if the playlist is reloaded). If you have a separate concept of playlist (currently selected songs for playing) and library (persistent metadata), this might make sense to think about, too.

**Comment by emoon** [comment]
Yes. I have both concepts currently (I store metadata in a sqlite3 "song db" which can be seen as a library) so this is a good idea.
